note to self CI is failing with
```>                   self.init_lib(args=['-c'], directory=tmpdir)
E                   TypeError: Model.init_lib() got an unexpected keyword argument 'args'
```

--- a/openmc/model/model.py
+++ b/openmc/model/model.py
@@ -2583,10 +2583,14 @@ class Model:
             # TODO: Can this be done without having to init/finalize?
             for univ in self.geometry.get_all_universes().values():
                 if isinstance(univ, openmc.DAGMCUniverse):
-                    # Initialize in volume calculation mode (non-transport mode)
+                    # Initialize in plotting mode (non-transport mode)
                     # This mode doesn't require
                     # valid transport settings like particles/batches
-                    self.init_lib(args=['-c'], directory=tmpdir)
+                    original_run_mode = self.settings.run_mode
+                    self.settings.run_mode = 'plot' 
+                    self.init_lib(directory=tmpdir)
+                    # Restore original run mode
+                    self.settings.run_mode = original_run_mode
                     self.sync_dagmc_universes()
                     self.finalize_lib()
                     break

--- a/openmc/model/model.py
+++ b/openmc/model/model.py
@@ -2583,11 +2583,11 @@ class Model:
             # TODO: Can this be done without having to init/finalize?
             for univ in self.geometry.get_all_universes().values():
                 if isinstance(univ, openmc.DAGMCUniverse):
-                    # Initialize in plotting mode (non-transport mode)
+                    # Initialize in stochastic volume mode (non-transport mode)
                     # This mode doesn't require
                     # valid transport settings like particles/batches
                     original_run_mode = self.settings.run_mode
-                    self.settings.run_mode = 'plot' 
+                    self.settings.run_mode = 'volume' 
                     self.init_lib(directory=tmpdir)
                     self.sync_dagmc_universes()
                     self.finalize_lib()

--- a/openmc/model/model.py
+++ b/openmc/model/model.py
@@ -2589,10 +2589,10 @@ class Model:
                     original_run_mode = self.settings.run_mode
                     self.settings.run_mode = 'plot' 
                     self.init_lib(directory=tmpdir)
-                    # Restore original run mode
-                    self.settings.run_mode = original_run_mode
                     self.sync_dagmc_universes()
                     self.finalize_lib()
+                    # Restore original run mode
+                    self.settings.run_mode = original_run_mode
                     break
 
             # Make sure all materials have a name, and that the name is a valid HDF5

--- a/tests/unit_tests/dagmc/test_convert_to_multigroup.py
+++ b/tests/unit_tests/dagmc/test_convert_to_multigroup.py
@@ -23,12 +23,21 @@ def test_convert_to_multigroup_without_particles_batches(run_in_tmpdir):
     # Use minimal tetrahedral DAGMC file
     dagmc_file = Path(__file__).parent / "dagmc_tetrahedral_no_graveyard.h5m"
     dagmc_univ = openmc.DAGMCUniverse(dagmc_file, auto_geom_ids=True)
+    bound_dagmc_univ = dagmc_univ.bounded_universe(padding_distance=1)
 
     # Create model WITHOUT setting particles or batches
     model = openmc.Model()
     model.materials = openmc.Materials([mat])
-    model.geometry = openmc.Geometry(dagmc_univ)
+    model.geometry = openmc.Geometry(bound_dagmc_univ)
     model.settings = openmc.Settings()  # Note: no particles or batches set!
+
+    model.settings.run_mode = 'fixed source'
+    
+    # Create a point source
+    my_source = openmc.IndependentSource()
+    my_source.space = openmc.stats.Point((0.25, 0.25, 0.25))
+    my_source.energy = openmc.stats.delta_function(14e6)
+    model.settings.source = my_source
 
     # This should work without requiring particles/batches to be set
     # convert_to_multigroup handles initialization internally using non-transport mode

